### PR TITLE
[BEAM-4277] Adding unshaded jars used by Dataflow to poms.

### PR DESCRIPTION
**Pull request by youngoli**
Dataflow needs unshaded jars created and still uses some of these poms.
Adding needed build rules to the poms.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand:
   - [x] What the pull request does
   - [x] Why it does it
   - [x] How it does it
   - [ ] Why this approach
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by youngoli** [comment]
R: @lukecwik, @tgroh, whoever gets around to it first

File: runners/direct-java/pom.xml [hunk 1]
```diff
@@ -125,6 +125,23 @@
             </execution>
           </executions>
         </plugin>
+        <!-- builds unshaded JARs used by Cloud Dataflow -->
```
Comment (tgroh) [inline runners/direct-java/pom.xml]: ?

Is this used by internal tests, or something? the Dataflow runtime shouldn't really be depending on the `DirectRunner`

Comment (youngoli) [reply runners/direct-java/pom.xml]: I believe that it's used by internal tests, yes.

**Comment by youngoli** [comment]
Run Java PreCommit

**Pull request merged**